src/fides/api/util/tcf/experience_meta.py

```diff
@@ -1,32 +1,49 @@
 import hashlib
 import json
-from typing import Dict, List
+from typing import Dict, List, Union
 
-from pydantic import Extra, root_validator
+from pydantic import Extra, Field, root_validator
 
 from fides.api.models.privacy_notice import UserConsentPreference
 from fides.api.schemas.base_class import FidesSchema
 from fides.api.schemas.privacy_experience import ExperienceMeta
-from fides.api.schemas.tcf import TCMobileData
+from fides.api.schemas.tcf import (
+    TCFVendorConsentRecord,
+    TCFVendorLegitimateInterestsRecord,
+    TCMobileData,
+)
 from fides.api.util.tcf.tc_mobile_data import build_tc_data_for_mobile
 from fides.api.util.tcf.tc_model import TCModel, convert_tcf_contents_to_tc_model
 from fides.api.util.tcf.tcf_experience_contents import TCFExperienceContents
 
 
 class TCFVersionHash(FidesSchema):
-    """Minimal subset of the TCF experience details that capture when consent should be resurfaced"""
+    """Minimal subset of the TCF experience details that capture when consent should be resurfaced
 
-    policy_version: int
-    purpose_consents: List[int]
-    purpose_legitimate_interests: List[int]
-    special_feature_optins: List[int]
-    vendor_consents: List[int]
-    vendor_legitimate_interests: List[int]
+    These values are later hashed, and you can treat a change in a hash as a trigger for re-establishing
+    transparency and consent
+    """
+
+    policy_version: int = Field(
+        description="TCF Policy Version. A new policy version invalidates existing strings and requires CMPs to "
+        "re-establish transparency and consent from users."
+    )
+    vendor_purpose_consents: List[str] = Field(
+        description="Stores vendor * purpose * consent legal basis in the format "
+        "<universal_vendor_id>-<comma separated purposes>"
+    )
+    vendor_purpose_legitimate_interests: List[str] = Field(
+        description="Stores vendor * purpose * legitimate interest legal basis in the format"
+        " <universal_vendor_id>-<comma separated purposes>"
+    )
+    gvl_vendors_disclosed: List[int] = Field(
+        description="List of GVL vendors disclosed from the last vendor list"
+    )
 
     @root_validator()
     @classmethod
     def sort_lists(cls, values: Dict) -> Dict:
-        """Verify lists are sorted ascending for repeatability"""
+        """Verify lists are sorted for repeatability"""
```

Comment:
This existing validator on the TCFVersionHash sorts the keys in each list. It's currently sorting them as strings, for example, "gacp.100" comes before "gacp.8-1".  Although this might not be how humans would expect this to be sorted, I am not implementing a more complex sorting strategy here because these are just the raw contents of the hash.

I experimented with custom sort functions that treated the letter and numerical sections separately, but was concerned about potential future vendor keys that might break this. I think the important thing is that it is sorted.
```
[
      "gacp.100",
      "gacp.8-1",
      "gvl.2-1,2,3,4,7,9,10",
      "gvl.8-1,3,4",
  ]
```

Reply:
this makes perfect sense and i think your decision is a wise one!

could add in a note
```suggestion
        """Verify lists are deterministically sorted for repeatability"""
```
but maybe that's overkill :)

Reply:
that's good 👍